src/tensorwaves/data/generate.py

```diff
@@ -12,64 +14,58 @@
 )
 from tensorwaves.interfaces import (
     Function,
-    Kinematics,
     PhaseSpaceGenerator,
     UniformRealNumberGenerator,
 )
-from tensorwaves.physics.helicity_formalism.kinematics import (
-    HelicityKinematics,
-    ParticleReactionKinematicsInfo,
-)
 
 
 def _generate_data_bunch(
     bunch_size: int,
     phsp_generator: PhaseSpaceGenerator,
     random_generator: UniformRealNumberGenerator,
     intensity: Function,
-    kinematics: Kinematics,
-) -> np.ndarray:
+    kinematics: HelicityKinematics,
+) -> Tuple[MomentumPool, float]:
     phsp_sample, weights = phsp_generator.generate(
         bunch_size, random_generator
     )
     dataset = kinematics.convert(phsp_sample)
     intensities = intensity(dataset)
-    maxvalue = np.max(intensities)
+    maxvalue: float = np.max(intensities)
 
     uniform_randoms = random_generator(bunch_size, max_value=maxvalue)
 
-    phsp_sample = phsp_sample.transpose(1, 0, 2)
-
-    return (phsp_sample[weights * intensities > uniform_randoms], maxvalue)
+    hit_and_miss_sample = phsp_sample.select_events(
+        weights * intensities > uniform_randoms
+    )
+    return hit_and_miss_sample, maxvalue
 
 
 def generate_data(
     size: int,
     kinematics: HelicityKinematics,
     intensity: Function,
     phsp_generator: Callable[
-        [ParticleReactionKinematicsInfo], PhaseSpaceGenerator
+        [ReactionInfo], PhaseSpaceGenerator
     ] = TFPhaseSpaceGenerator,
     random_generator: Optional[UniformRealNumberGenerator] = None,
     bunch_size: int = 50000,
-) -> np.ndarray:
+) -> MomentumPool:
```

Comment:
I'm not a huge fan of the choice `MomentumPool` here (I know this was kind of brought to live by myself) My reasoning is that Pool suggest some kind of design (https://en.wikipedia.org/wiki/Pool_(computer_science)) and here its simply a dictionary of momenta or something in that direction

Reply:
Fixed by 6b8a559 and follow-up commits as well as a rename from `MomentumPool` to `EventCollection` in https://github.com/ComPWA/expertsystem/pull/454